source/features/prevent-link-loss.tsx

```diff
@@ -63,12 +63,11 @@ const updateUI = debounceFn(({delegateTarget: field}: delegate.Event<Event, HTML
 	wait: 300,
 });
 
-function init(): Deinit {
-	return [
-		updateUI.cancel,
-		delegate(document, 'form:is(#new_issue, #new_release) textarea, form.js-new-comment-form textarea, textarea.comment-form-textarea', 'input', updateUI),
-		delegate(document, '.rgh-prevent-link-loss', 'click', handleButtonClick),
-	];
+function init(signal: AbortSignal): Deinit {
+	delegate(document, 'form:is(#new_issue, #new_release) textarea, form.js-new-comment-form textarea, textarea.comment-form-textarea', 'input', updateUI, {signal});
+	delegate(document, '.rgh-prevent-link-loss', 'click', handleButtonClick, {signal});
+
+	return updateUI.cancel;
```

Comment:
Thoughts?

```suggestion
	signal.addEventListener('abort', updateUI.cancel);
```

I'd love to have an `onAbort(emitter: signal, listener: function | controller)` helper for this kind of situations:

```suggestion
	onAbort(signal, updateUI.cancel)
```

and also when we have multiple controllers and we want to link them together, like:


```js
function foo(otherSignal) {
	const controller = new AbortController
	onAbort(otherSignal, controller)
	return controller;
}
```

In this case, `controller` would also be aborted when `otherSignal` is aborted.

I'm not sure it's useful in this specific case though, `return fn` will always be more convenient.

Reply:
For this specific case we should just drop this. There's no chance that the user changes page within 300ms of the last keypress